### PR TITLE
Added some validations to jobs-create page

### LINKED ISSUE #1140: Validation is absent in the Jobs Create page
## Context

Validation is absent in the `/jobs/create` page. (in the form specifically)

## Expected Behavior

There should be some kind of validation available for the required fields.

## Current Behavior

No validation error is shown in UI. (which should not be the case)

## Screenshots

Screen shot of page

## Steps to reproduce

Please provide detailed steps for reproducing the issue

1. Go to `/jobs/create` page and put random values in the fields
2. Click on `Submit and checkout` button
3. No validation error is shown.

### PR DESCRIPTION
**Pull request by Nil2000**
# ✨ Codu Pull Request 💻

Fixes #1140 

## Pull Request details

- Validation part attached

## Any Breaking changes

- None

## Associated Screenshots

image
image

### CODE REVIEW

File: schema/job.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    .string()
+    .min(1, "Company name should contain atleast 1 character")
+    .max(50, "Company name should contain atmost 50 characters"),
+  jobTitle: z
+    .string()
+    .min(3, "Job title should contain atleast 3 character")
+    .max(50, "Job title should contain atmost 50 characters"),
+  jobDescription: z
+    .string()
+    .max(200, "Job Description should contain atmost 200 characters")
```
Comment (NiallJoeMaher) [inline schema/job.ts]: We should have a min character count on this too. Without a job description the post will be pretty bad. Say 100 characters. you can max it at 2000.

File: schema/job.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  applicationUrl: z
+    .string()
+    .url("Provide a valid url")
+    .optional()
+    .or(z.literal("")),
+  remote: z.boolean().optional().default(false),
+  relocation: z.boolean().optional().default(false),
+  visa: z.boolean().optional().default(false),
+  jobType: z.enum(["full-time", "part-time", "freelancer", "other"]),
+});
```
Comment (NiallJoeMaher) [inline schema/job.ts:30]: Would love some thoughts on if you think all of these are sensible too? 🦾

Comment (Nil2000) [reply schema/job.ts:30]: I think yes, these should also be populated. As a result these values can be shown in the `jobs` page with their associated jobs.
A sample shown below present on [YC jobs page](https://www.workatastartup.com/companies?demographic=any&hasEquity=any&hasSalary=any&industry=any&interviewProcess=any&jobType=fulltime&layout=list-compact&role=eng&sortBy=created_desc&tab=any&usVisaNotRequired=any)

image

File: schema/job.ts (hunk 1 above)
Comment (Nil2000) [reply schema/job.ts]: Done ✅

**Comment by Nil2000** [comment]
@NiallJoeMaher @John-Paul-Larkin  Can I implement the upload image functionality here (similar to settings page)?

**Comment by NiallJoeMaher** [comment]
Yes, in fact you could do it with an action similar to how I handle the upload on the blog post. I'll share a link when I'm on my laptop.

**Review by NiallJoeMaher — APPROVED** [review]
Great work! You can work on the image upload next. Create the issue and I'll
Assign you.

**Pull request merged**